src/tools/cephfs/cephfs_shell.py

```diff
@@ -0,0 +1,844 @@
+                    if dir_name == '':
+                        dir_name = '/'
+                    items = [item for item in list_items(dir_name) for i in all_items if i.rsplit('/', 1)[1] == item.d_name.decode('utf-8') and not item.is_dir()]
+                    dirs = [(dir_name + '/' + item.d_name.decode('utf-8')).replace('//', '/') for item in list_items(dir_name) for i in all_items if i.rsplit('/', 1)[1] == item.d_name.decode('utf-8') and item.is_dir()]
+                    directories.extend(dirs)
+                    if len(dirs) == 0 :
+                        print(dir_name, ':')
+                    items = sorted(items, key = lambda item : item.d_name)
+                else :
+                    if dir_name != '' and dir_name != cephfs.getcwd().decode('utf-8'):
+                        print(dir_name, ':')
+                    items = sorted(list_items(dir_name), key = lambda item : item.d_name)
+                if len(items) > 2 and items[0].d_name.decode('utf-8') == '.':
+                    items = items[2:]
+                flag = 0
+                if args.S :
+                    items = sorted(items, key = lambda item : cephfs.stat(to_bytes(dir_name + '/' + item.d_name.decode('utf-8'))).st_size)    
+                if args.reverse :
+                    items = reversed(items)
+                for item in items:
+                    path = item
+                    if not isinstance(item, str) :
+                        path = item.d_name.decode('utf-8') 
+                        if item.is_dir() :
+                            flag = 1
+                        else :
+                            flag = 0
+                    if args.long and args.H  :
+                        print_long(cephfs.getcwd().decode('utf-8') + dir_name + '/' + path, flag, True)
+                    elif args.long :
+                        print_long(cephfs.getcwd().decode('utf-8') + dir_name + '/' + path, flag, False)
+                    else :
+                        values.append(colorama.Fore.BLUE * flag + path + '/'* flag + colorama.Style.RESET_ALL  * flag)
+                if not args.long :
+                    print_list(values, shutil.get_terminal_size().columns)
+                    if dir_name != directories[-1]:
+                        print()
+        except :
+            traceback.print_exc(file = sys.stdout)
+   
+    def complete_rmdir(self, text, line, begidx, endidx):
+        """ 
+        auto complete of file name.
+        """    
+        return self.complete_filenames(text, line, begidx, endidx)
+
+    rmdir_parser = argparse.ArgumentParser(description = 'Remove Directory.')
+    rmdir_parser.add_argument('dir_paths', help = 'Directory Path.',nargs = '*', default = [''])
+    rmdir_parser.add_argument('-p', '--parent', action = 'store_true', help = 'Remove parent directories as necessary. When this option is specified, no error is reported if a directory has any sub-directories, files')
```

Comment:
I'd prefer `rmdir` remain dumb and only removes empty directories. Let's add a recursive `-r` option to the `rm` command (just like the UNIX command).